runtime/commonMain/src/kotlinx/serialization/json/internal/ContextValidator.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json.internal
+
+import kotlinx.serialization.*
+import kotlinx.serialization.modules.*
+import kotlin.reflect.*
+
+internal class ContextValidator(private val discriminator: String) : SerialModuleCollector {
+    override fun <T : Any> contextual(kClass: KClass<T>, serializer: KSerializer<T>) {
+        // Nothing here
+    }
+
+    override fun <Base : Any, Sub : Base> polymorphic(
+        baseClass: KClass<Base>,
+        actualClass: KClass<Sub>,
+        actualSerializer: KSerializer<Sub>
+    ) {
+        val descriptor = actualSerializer.descriptor
+        for (i in 0 until descriptor.elementsCount) {
```

Comment:
Can be simplified to `discriminator in descriptor.elementNames()` then?

Reply:
It will create an undesired footprint. 
I will create `cachedElementNames` separately and left `elementNames` as is, so users won't be able to increase footprint for no real reason